os/arch/arm/Kconfig

```diff
@@ -271,6 +271,16 @@ config ARCH_DABORTSTACK
 		will be used during data abort handling. Recommended data abort stack
 		size is 1K.
 
+config MPU_STACK_OVERFLOW_PROTECTION
+	bool "MPU Stack Overflow Protection"
+	depends on BUILD_PROTECTED
```

Comment:
Can we support mpu stackoverflow protection with flat build?

Reply:
in flat build kernel is in privilege mode, so we need to disable it and add MPU to kernel region.
I will explore the feasibility for this.